lazyflow/operators/opCache.py

```diff
@@ -54,8 +54,11 @@ class MyCachingOperator(Cache, Operator):
     cleanup could occur while the cache is still under construction!
     """
 
-    def registerWithMemoryManager(self):
-        manager = CacheMemoryManager()
+    def registerWithMemoryManager(self, memory_manager=None):
```

Comment:
what is the use-case for passing a memory manager? Just testing, or do you also have something else in mind?

Reply:
Just testing, do you we should change this somehow maybe add underscore to indicate that it is an argumetn for internal use only?

Reply:
could we maybe make this a private method (with an underscore) and a mandatory `memory_manager` argument and a public one that just passes `cacheMemoryManager` to the private one?

Reply:
Problem with this approach that it's called from constructors of Op*Cache. See usage in tests https://github.com/ilastik/lazyflow/blob/b7daf4f5198c7c8d8d3df4b3ebdd8c2027e0e0c0/tests/test_lazyflow/test_operators/testCacheMemoryManager.py#L174 so as a result each cache would need to know to call this private method or we would have two managers with different policies managing the cache.

Reply:
how about not exposing this at all and using pytests monkeypatch functionality to inject a cachememorymanager during testing?

Reply:
> how about not exposing this at all and using pytests monkeypatch functionality to inject a cachememorymanager during testing?

Yep, great idea :)

Reply:
Is implicit monkey-patching really better than adding an explicit argument in this case? `OpCache` might as well work with anything that looks like a memory manager, not just a single global one.

Reply:
unless that there is a good reason for using a _specific_ different CacheMemoryManager I'd prefer to send a clear message to user of this class.

Reply:
Testing seems like a good reason since it frequently uncovers hidden dependencies. It seems strange to monkey-patch something that is under our control.

As for the clear message, we can add a docstring; if someone wants to use this param, they are expected to read docs anyway. Also, if you decide to leave this as a param, it might be a good idea to make it keyword-only.

Reply:
I agree with monkey patching approach because figuring out the best way to do dependency injection is out of the scope of this pull-request. The purpose of this pull-requests is only to enable to run lazyflow tests together with ilastik.